sales/sales/sales.py

```diff
@@ -0,0 +1,368 @@
+                    job=self.job,
+                    salary=self.salary,
+                )
+            )
+            session.commit()
+        return rx.window_alert(f"User {self.customer_name} has been added.")
+
+    def customer_page(self):
+        """The customer page."""
+        return rx.redirect("/")
+
+    def onboarding_page(self):
+        """The onboarding page."""
+        return rx.redirect("/onboarding")
+
+    def delete_customer(self, email: str):
+        """Delete a customer from the database."""
+        with rx.session() as session:
+            customer = session.exec(
+                select(Customer).where(Customer.email == email)
+            ).first()
+            session.delete(customer)
+            session.commit()
+
+    generate_email_data: dict = {}
+
+    async def call_openai(self):
+        name: str = self.generate_email_data["name"]
+        email: str = self.generate_email_data["email"]
+        age: int = self.generate_email_data["age"]
+        gender: str = self.generate_email_data["gender"]
+        location: str = self.generate_email_data["location"]
+        job: str = self.generate_email_data["job"]
+        salary: int = self.generate_email_data["salary"]
+        response = OpenAI().completions.create(
+            model="gpt-3.5-turbo-instruct",
+            prompt=f"Based on these {products} write a sales email to {name} and email {email} who is {age} years old and a {gender} gender. {name} lives in {location} and works as a {job} and earns {salary} per year. Make sure the email recommends one product only and is personalized to {name}. The company is named Reflex its website is https://reflex.dev",
+            temperature=0.7,
+            max_tokens=2250,
+            top_p=1,
+            frequency_penalty=0,
+            presence_penalty=0,
+        )
+        self.gen_response = False
+        # save the data related to email_content
+        self.email_content_data = response.choices[0].text
+        # update layout of email_content manually
+        return rx.set_value("email_content", self.email_content_data)
+
+    def generate_email(
+        self,
+        name: str,
+        email: str,
+        age: int,
+        gender: str,
+        location: str,
+        job: str,
+        salary: int,
+    ):
+        self.generate_email_data["name"] = name
+        self.generate_email_data["email"] = email
+        self.generate_email_data["age"] = age
+        self.generate_email_data["gender"] = gender
+        self.generate_email_data["location"] = location
+        self.generate_email_data["job"] = job
+        self.generate_email_data["salary"] = salary
+        self.text_area_disabled = True
+        self.gen_response = True
+        return State.call_openai
+
+    @rx.var
+    def get_users(self) -> list[Customer]:
+        """Get all users from the database."""
+        with rx.session() as session:
+            return session.exec(select(Customer)).all()
```

Comment:
i think we should make this an `on_load`

```suggestion
    def get_users(self) -> list[Customer]:
        """Get all users from the database."""
        with rx.session() as session:
            self.users = session.exec(select(Customer)).all()
```

Then update `add_customer` and `delete_customer` to call this afterwards. And have the `/` route call it `on_load`.

This tends to be better for data apps, because it avoids database queries on every action, and allows the app to export without a database defined (which is why the CI is failing)